src/common/Button/Button.tsx

```diff
@@ -1,12 +1,22 @@
-import * as types from 'src/types';
-
 import { FC } from 'react';
 
 import styles from './Button.module.css';
 
-export const Button: FC<types.ButtonProps> = ({ text, onClick }) => {
+interface ButtonProps {
+	text: string;
+	onClick?(
+		data?: string | React.MouseEvent
+	): void | React.MouseEventHandler<HTMLButtonElement> | Promise<void>;
+	type?: 'button' | 'submit' | 'reset';
+}
+
+export const Button: FC<ButtonProps> = ({
+	text,
+	onClick = undefined,
```

Comment:
`onClick` is undefined if it's not passed, so `= undefined` can be removed

```suggestion
	onClick,
```